plutus/budget_manager/mysql.py

```diff
@@ -0,0 +1,162 @@
+from constants import APP
+import datetime
+from googleapiclient import discovery
+import google.auth
+import json
+import logging
+import markus
+from pprint import pprint
+from protobuf_to_dict import protobuf_to_dict
+import pymysql
+import re
+import sys
+
+log = logging.getLogger(APP + '.mysql')
+metrics = markus.get_metrics(APP + '.mysql')
+
+def upsert_budget(mysql_cursor, budget, project_id, config_type, alert_emails):
+    ''' Upserts a row in the budgets mysql table. '''
+
+    budget_dict = protobuf_to_dict(budget, use_enum_labels=True)
+
+    # Construct column values
+    budget_id = budget_dict['name']
+    display_name = budget_dict['display_name']
+
+    # Currently GCP Budgets API limits to either 0 or 1 project assigned to a budget (0 meaning all projects)
+    if ('projects' in budget_dict['budget_filter'] and len(budget_dict['budget_filter']['projects']) == 1):
+        project_number = budget_dict['budget_filter']['projects'][0].replace("projects/", "")
+    else:
+        log.error("Unexpected error. Budgets API currently expects budget_filter.projects to be either 0 or 1 projects.")
+        log.error("Unless this has changed. Also, plutus only currently manages budgets for single projects.")
```

Comment:
These two log lines should be consolidated and a bit verbose. Maybe move this into comments and have a short explanation here instead?